test/unit/template_importer_test.rb

```diff
@@ -142,5 +142,16 @@ def self.import!(name, text, _metadata)
       assert_equal File.read(get_template('ptable1.erb')), pt.layout
       assert_equal 'Debian', pt.os_family
     end
+
+    test 'purge! removes expected templates' do
+      FactoryGirl.create(:provisioning_template, :name => 'FooBar delete_me')
+      FactoryGirl.create(:provisioning_template, :name => 'FooBar delete_me_too')
+      FactoryGirl.create(:provisioning_template, :name => 'keep_me')
+      count = ProvisioningTemplate.all.size
+      @importer.purge!
+      refute ProvisioningTemplate.find_by_name('FooBar %').present?
+      assert ProvisioningTemplate.find_by_name('keep_me').present?
+      assert ProvisioningTemplate.all.size == count - 2
+    end # 'purge! removes expected template'
```

Comment:
You almost had it, just one character out :)

See https://github.com/theforeman/foreman_templates/blob/master/test/unit/template_importer_test.rb#L81 for an example, but it's expecting an options hash, so you want `:negate => true`